salvo/src/lib/benchmark/test_fully_dockerized_benchmark.py

```diff
@@ -29,34 +85,166 @@ def test_source_to_build_envoy():
   Validate that sources are defined that enable us to build the Envoy image
   We do not expect the validation logic to throw an exception
   """
-  pass
+  # create a valid configuration with a missing Envoy image
+  job_control = JobControl()
```

Comment:
This test setup code is fairly verbose and looks repetitive. Unless there are significant differences between these test cases, we should consider extracting it to a helper function. As it stands, it distracts the reader from the core concepts of the test cases.

We could have a function that returns the protos setup for the general case and then simply adjust what we need in each test case. That should also make it easier for the reader to spot the "differences" between the test cases which are hard to identify in the current version.

Reply:
Yep.  Will update.